### PR TITLE
fix(cookies): be able to split `set-cookie` without `getAll`

### PR DESCRIPTION
**Pull request by balazsorban44**
Currently, if a user sets multiple cookies in a series, only the second cookie is set in the `set-cookie` header.

When this is merged, I will update https://github.com/vercel/next.js/tree/canary/packages/next/src/server/web/spec-extension/cookies to drop the code and use the pre-compiled version of this package again.

[Slack thread 1](https://vercel.slack.com/archives/C03S8ED1DKM/p1675694485304329), [Slack thread 2](https://vercel.slack.com/archives/C03ENM5HB4K/p1668173326539069)

### CODE REVIEW

File: packages/cookies/src/response-cookies.ts [hunk 1]
```diff
@@ -14,14 +14,20 @@ export class ResponseCookies {
...
 
-    for (const header of headers) {
-      const parsed = parseSetCookieString(header)
-      if (parsed) {
-        this._parsed.set(parsed.name, parsed)
-      }
+    const setCookie =
+      // @ts-expect-error See https://github.com/whatwg/fetch/issues/973
+      responseHeaders.getAll?.('set-cookie') ??
+      responseHeaders.get('set-cookie') ??
```
Comment (feugy) [inline packages/cookies/src/response-cookies.ts:21]: **praise:** Thanks for removing this hard dependency on `getAll()`!! I banged onto it while working on isomorphic, but I had to stop for changing priorities. Thanks a million!

File: packages/cookies/src/response-cookies.ts [hunk 2]
```diff
@@ -102,3 +108,82 @@ function normalizeCookie(cookie: ResponseCookie = { name: '', value: '' }) {
...
+ * Set-Cookie header field-values are sometimes comma joined in one string. This splits them without choking on commas
+ * that are within a single set-cookie field-value, such as in the Expires portion.
+ * This is uncommon, but explicitly allowed - see https://tools.ietf.org/html/rfc2616#section-4.2
+ * Node.js does this for every header *except* set-cookie - see https://github.com/nodejs/node/blob/d5e363b77ebaf1caf67cd7528224b651c86815c1/lib/_http_incoming.js#L128
+ * React Native's fetch does this for *every* header, including set-cookie.
+ *
+ * Based on: https://github.com/google/j2objc/commit/16820fdbc8f76ca0c33472810ce0cb03d20efe25
+ * Credits to: https://github.com/tomball for original and https://github.com/chrusart for JavaScript implementation
+ */
+function splitCookiesString(cookiesString: string) {
```
Comment (feugy) [inline packages/cookies/src/response-cookies.ts:124]: **suggestion:** we already have this function in `packages/node-utils/src/edge-to-node/headers`. Can we share it between the two packages? maybe a third package called `internals-utils`? @Kikobeats, @javivelasco, what do you think?

**Review by feugy — COMMENTED** [review]
Works well!

**Review by feugy — APPROVED** [review]
Seen with @Kikobeats, let's merge this as is, we'll do the code sharing in subsequent work.

Thanks again!

File: packages/cookies/src/response-cookies.ts (hunk 2 above)
Comment (Kikobeats) [reply packages/cookies/src/response-cookies.ts:124]: yes, we can create a package for that, also for testing it properly.

**Review by Kikobeats — APPROVED** [review]

**Pull request merged**